casper/src/test/scala/coop/rchain/casper/util/rholang/Interactive.scala

```diff
@@ -53,7 +55,7 @@ class Interactive private (runtime: Runtime[Task])(implicit scheduler: Scheduler
   def tuplespace: String = StoragePrinter.prettyPrint(runtime.space.store)
 
   def eval(code: String): Unit = {
-    TestUtil.eval(code, runtime)
+    TestUtil.eval(code, runtime).runSyncUnsafe(Duration.Inf)
```

Comment:
This may cause tests to hang in case of deadlocks. Better use a finite duration

Reply:
This isn't used for tests, but for interactive usage

Reply:
I see! In that case instead of introducing yet another `unsafe` operation I would opt for introducing a for comprehension here

Reply:
The unsafe was always there, under the `Unit` returned previously. I'd fix it if I was going to work with that code soon, but I don't think so :)